app/controllers/users_controller.rb

```diff
@@ -21,7 +21,7 @@ def index
     user_type_conditions = []
 
     if params[:portal_admin].to_s.length > 0
-      user_type_conditions << 'users.site_admin = true'
+      user_type_conditions << 'roles_users.role_id = "1"'
```

Comment:
this is hard coding the id of the site_admin role. In theory this id could be anything. So instead this id should be looked up. Something like `admin_id = Role.where(title: 'admin').first.id`

Reply:
Ah, right. Understood. How does it look now?